src/components/DataView/DataView.js

```diff
@@ -0,0 +1,313 @@
+
+  const listMode =
+    mode === 'list' || (mode !== 'table' && layoutName === 'small')
+
+  const { allSelected, selectAll, toggleEntry, selectedIndexes } = useSelection(
+    entries,
+    onSelectEntries
+  )
+
+  const hasAnyActions = Boolean(renderEntryActions)
+  const hasAnyChild = Boolean(renderEntryChild)
+  const canSelect = Boolean(onSelectEntries)
+
+  const pages = Math.ceil(entries.length / entriesPerPage)
+
+  const displayFrom = entriesPerPage * selectedPage
+  const displayTo = displayFrom + entriesPerPage
+  const displayedEntries = prepareEntries(
+    entries,
+    displayFrom,
+    displayTo,
+    selectedIndexes
+  )
+
+  const preparedFields = prepareFields(fields)
+  const renderedEntries = renderEntries(displayedEntries, {
+    fields,
+    renderEntry,
+    renderEntryActions,
+    renderEntryChild,
+    mode: listMode ? 'list' : 'table',
+  })
+
+  const alignChildOnField = fields.findIndex(field => field.childStart)
+
+  return (
+    <Box padding={false}>
+      {heading && (
+        <div
+          css={`
+            padding: ${2 * GU}px ${3 * GU}px;
+          `}
+        >
+          {typeof heading === 'string' ? (
+            <h1
+              css={`
+                font-size: 16px;
+                font-weight: 400;
+                margin-bottom: ${2 * GU}px;
+              `}
+            >
+              {heading}
+            </h1>
+          ) : (
+            heading
+          )}
+        </div>
+      )}
+
+      {listMode ? (
+        <ListView
+          allSelected={allSelected}
+          entries={renderedEntries}
+          fields={preparedFields}
+          hasAnyChild={hasAnyChild}
+          onSelect={toggleEntry}
+          onSelectAll={selectAll}
+          renderSelectionCount={renderSelectionCount}
+          selectable={canSelect}
+        />
+      ) : (
+        <TableView
+          alignChildOnField={Math.min(
+            Math.max(-1, alignChildOnField),
+            fields.length - 1
+          )}
+          allSelected={allSelected}
+          entries={renderedEntries}
+          fields={preparedFields}
+          hasAnyActions={hasAnyActions}
+          hasAnyChild={hasAnyChild}
+          onSelect={toggleEntry}
+          onSelectAll={selectAll}
+          renderSelectionCount={renderSelectionCount}
+          rowHeight={tableRowHeight}
+          selectable={canSelect}
+          selectedCount={selectedIndexes.length}
+        />
+      )}
+
+      {pages > 1 && (
+        <div
+          css={`
+            border-top: 1px solid ${theme.border};
+          `}
+        >
+          <Pagination
+            pages={pages}
+            selected={selectedPage}
+            onChange={handlePageChange}
+            touchMode={layoutName === 'small'}
+          />
+        </div>
+      )}
+    </Box>
+  )
+})
+
+DataView.propTypes = {
+  page: PropTypes.number,
+  entries: PropTypes.array.isRequired,
+  entriesPerPage: PropTypes.number,
+  fields: PropTypes.array,
```

Comment:
```suggestion
  fields: PropTypes.array.isRequired,
```

Not providing `fields` will cause an error